web/src/sidebar_ui.ts

```diff
@@ -294,18 +294,61 @@ export function initialize_right_sidebar(): void {
         buddy_list.toggle_other_users_section();
     });
 
+    function close_buddy_list_popover(): void {
+        if (popover_menus.popover_instances.buddy_list !== null) {
+            popover_menus.popover_instances.buddy_list.destroy();
+            popover_menus.popover_instances.buddy_list = null;
+        }
+    }
+
     popover_menus.register_popover_menu("#buddy-list-menu-icon", {
         theme: "popover-menu",
         placement: "right",
         onCreate(instance) {
             popover_menus.popover_instances.buddy_list = instance;
-            instance.setContent(ui_util.parse_html(render_buddy_list_popover()));
+            instance.setContent(
+                ui_util.parse_html(
+                    render_buddy_list_popover({
+                        display_style_options: settings_config.user_list_style_values,
+                        can_invite_users:
+                            settings_data.user_can_invite_users_by_email() ||
+                            settings_data.user_can_create_multiuse_invite(),
+                    }),
+                ),
+            );
+        },
+        onMount() {
+            const current_user_list_style =
+                settings_preferences.user_settings_panel.settings_object.user_list_style;
+            $("#buddy-list-actions-menu-popover")
+                .find(`.user_list_style_choice[value=${current_user_list_style}]`)
+                .prop("checked", true);
         },
         onHidden() {
-            if (popover_menus.popover_instances.buddy_list !== null) {
-                popover_menus.popover_instances.buddy_list.destroy();
-                popover_menus.popover_instances.buddy_list = null;
-            }
+            close_buddy_list_popover();
         },
     });
+
+    $("body").on(
+        "click",
+        "#buddy-list-actions-menu-popover .display-style-selector",
+        function (this: HTMLElement) {
+            const data = {user_list_style: $(this).val()};
+            const current_user_list_style =
+                settings_preferences.user_settings_panel.settings_object.user_list_style;
+
+            if (current_user_list_style === data.user_list_style) {
+                close_buddy_list_popover();
+                return;
+            }
+
+            void channel.patch({
+                url: "/json/settings",
+                data,
+                success() {
+                    close_buddy_list_popover();
+                },
```

Comment:
I guess we don't have a good place to display an error if this fails.

Reply:
Oh yeah, I was thinking about that while working on it and forgot to leave a note here. But that's the conclusion I came to as well.

Reply:
Worth having an empty `error` handler with a comment for that sort of detail. I think in this case I'm tempted to be lazy and leave it as a noop, but I guess `feedback_widget` might be suitable and perhaps not hard to hook up?